### ISSUE TITLE
Merges are not triggering a request?

### ISSUE THREAD

**Issue by yuniorb** [tomasbjerre/pull-request-notifier-for-bitbucket#195]
## Resolution
After digging into the code and the plugin settings this turned out to be a configuration problem on my side. It turns out that the `state` section: 
![2017-02-12_01-20-33](https://cloud.githubusercontent.com/assets/12397873/22860108/7d1476d0-f0c1-11e6-9374-a83ec59f9122.png)
doesn't trigger the DECLINED or MERGED events if you select them. Every other option is pretty intuitive and you can deduce what it does, but the `state` section might benefit from some docs/clarification.

## Issue
Merge trigger doesn't seem to be working. I am using request bin with the everything url in the body and using a POST. I searched the logs and was able to find that there is an entry generated when requests are sent from `prnfb` out to the url. Here is the log I found for a comment added event (all requests were made with the same configuration detailed below).
```
2017-02-11 23:54:15,625 INFO  [threadpool:thread-1] foo-user @Q6ZUU2x1434x19x0 m61y9t 0:0:0:0:0:0:0:1 "POST /rest/api/latest/projects/ABC/repos/foo-repo/pull-requests/7/comments HTTP/1.1" se.bjurr.prnfb.http.UrlInvoker Url: "http://requestb.in/foobar"
```
Below is the log generated when the PR was merged. I couldn't find a similar entry calling out to requestb.in in this case. Any ideas what could be the problem? Is it configuration, or an issue? 

```
2017-02-11 23:58:08,243 DEBUG [http-nio-7990-exec-4] foo-user @Q6ZUU2x1438x22x0 m61y9t 0:0:0:0:0:0:0:1 "POST /rest/api/latest/projects/ABC/repos/foo-repo/pull-requests/7/merge HTTP/1.1" c.a.s.i.scm.git.merge.MergeCommand ABC/foo-repo[1]: Merging ABC/foo-repo refs/heads/jr -> refs/heads/master in /Users/foo-user/Atlassian/ApplicationData/Bitbucket/tmp/git/foo-repo-work6081183366483310352.git
2017-02-11 23:58:08,302 DEBUG [http-nio-7990-exec-4] foo-user @Q6ZUU2x1438x22x0 m61y9t 0:0:0:0:0:0:0:1 "POST /rest/api/latest/projects/ABC/repos/foo-repo/pull-requests/7/merge HTTP/1.1" c.a.bitbucket.scm.BaseCommand Executed /usr/bin/git merge -m Merge pull request #7 in ABC/foo-repo from jr to master --no-ff --log --allow-unrelated-histories 9cf96ea3a1db6021f0f423cef9005584b107a055
2017-02-11 23:58:08,302 DEBUG [http-nio-7990-exec-4] foo-user @Q6ZUU2x1438x22x0 m61y9t 0:0:0:0:0:0:0:1 "POST /rest/api/latest/projects/ABC/repos/foo-repo/pull-requests/7/merge HTTP/1.1" c.a.s.i.scm.git.merge.MergeCommand ABC/foo-repo[1]: Fetching successful merge to refs/heads/master from /Users/foo-user/Atlassian/ApplicationData/Bitbucket/tmp/git/foo-repo-work6081183366483310352.git
2017-02-11 23:58:08,302 DEBUG [http-nio-7990-exec-4] foo-user @Q6ZUU2x1438x22x0 m61y9t 0:0:0:0:0:0:0:1 "POST /rest/api/latest/projects/ABC/repos/foo-repo/pull-requests/7/merge HTTP/1.1" c.a.s.i.s.g.m.ObjectFetchStrategy ABC/foo-repo refs/heads/jr -> refs/heads/master: Fetching single-repository merge by copying new objects
2017-02-11 23:58:08,336 DEBUG [http-nio-7990-exec-4] foo-user @Q6ZUU2x1438x22x0 m61y9t 0:0:0:0:0:0:0:1 "POST /rest/api/latest/projects/ABC/repos/foo-repo/pull-requests/7/merge HTTP/1.1" c.a.bitbucket.scm.BaseCommand Executed /usr/bin/git update-ref refs/heads/master 2539c068ff9c7ba6ada260912e59aeef60cda8b8 cc2c00f9b641887ca101e1d5436f8c8ad5a8e9ad
2017-02-11 23:58:08,357 DEBUG [http-nio-7990-exec-4] foo-user @Q6ZUU2x1438x22x0 m61y9t 0:0:0:0:0:0:0:1 "POST /rest/api/latest/projects/ABC/repos/foo-repo/pull-requests/7/merge HTTP/1.1" c.a.s.i.e.TransactionAwareEventPublisher Deferring publishing for PullRequestMergeActivityEvent until AFTER_COMMIT
2017-02-11 23:58:08,360 DEBUG [http-nio-7990-exec-4] foo-user @Q6ZUU2x1438x22x0 m61y9t 0:0:0:0:0:0:0:1 "POST /rest/api/latest/projects/ABC/repos/foo-repo/pull-requests/7/merge HTTP/1.1" c.a.s.i.e.TransactionAwareEventPublisher Deferring publishing for AnalyticsPullRequestMergedEvent until AFTER_COMMIT
2017-02-11 23:58:08,365 DEBUG [http-nio-7990-exec-4] foo-user @Q6ZUU2x1438x22x0 m61y9t 0:0:0:0:0:0:0:1 "POST /rest/api/latest/projects/ABC/repos/foo-repo/pull-requests/7/merge HTTP/1.1" c.a.s.i.e.TransactionAwareEventPublisher Publishing PullRequestMergeActivityEvent after commit
2017-02-11 23:58:08,365 DEBUG [http-nio-7990-exec-4] foo-user @Q6ZUU2x1438x22x0 m61y9t 0:0:0:0:0:0:0:1 "POST /rest/api/latest/projects/ABC/repos/foo-repo/pull-requests/7/merge HTTP/1.1" c.a.s.i.e.TransactionAwareEventPublisher Publishing AnalyticsPullRequestMergedEvent after commit
2017-02-11 23:58:08,366 DEBUG [AtlassianEvent::thread-3] foo-user @Q6ZUU2x1438x22x0 m61y9t 0:0:0:0:0:0:0:1 "POST /rest/api/latest/projects/ABC/repos/foo-repo/pull-requests/7/merge HTTP/1.1" c.a.s.i.n.DefaultNotificationManager Managing notification: PullRequestActivityNotification{user=InternalNormalUser{id=1, username=foo-user}, timestamp=Sat Feb 11 23:58:08 EST 2017, pullRequest=foo-repo:7, activity=InternalPullRequestMergeActivity{id=77, user=InternalNormalUser{id=1, username=foo-user}, action=MERGED, pullRequest.id=7}}
2017-02-11 23:58:08,367 DEBUG [AtlassianEvent::thread-4] foo-user @Q6ZUU2x1438x22x0 m61y9t 0:0:0:0:0:0:0:1 "POST /rest/api/latest/projects/ABC/repos/foo-repo/pull-requests/7/merge HTTP/1.1" c.a.s.i.n.DefaultNotificationManager Managing notification: PullRequestMergedNotification{user=InternalNormalUser{id=1, username=foo-user}, timestamp=Sat Feb 11 23:58:08 EST 2017, pullRequest=foo-repo:7}
2017-02-11 23:58:08,430 DEBUG [ref-sync:thread-1] foo-user @Q6ZUU2x1438x22x0 m61y9t 0:0:0:0:0:0:0:1 "POST /rest/api/latest/projects/ABC/repos/foo-repo/pull-requests/7/merge HTTP/1.1" c.a.s.i.r.s.auto.AutoRefSyncListener ABC/foo-repo[1]: Not updating status; synchronization is either not enabled or not available
2017-02-11 23:58:08,456 DEBUG [AtlassianEvent::thread-2] foo-user @Q6ZUU2x1438x22x0 m61y9t 0:0:0:0:0:0:0:1 "POST /rest/api/latest/projects/ABC/repos/foo-repo/pull-requests/7/merge HTTP/1.1" c.a.bitbucket.scm.BaseCommand Executed /usr/bin/git cat-file -p refs/heads/master:.mailmap
2017-02-11 23:58:08,477 DEBUG [AtlassianEvent::thread-6] foo-user @Q6ZUU2x1438x22x0 m61y9t 0:0:0:0:0:0:0:1 "POST /rest/api/latest/projects/ABC/repos/foo-repo/pull-requests/7/merge HTTP/1.1" c.a.bitbucket.scm.BaseCommand Executed /usr/bin/git rev-list --format=%H%x02%P%x02%aN%x02%aE%x02%at%n%B%n%x03END%x04 --ignore-missing --no-min-parents --stdin --
2017-02-11 23:58:08,477 DEBUG [AtlassianEvent::thread-4] foo-user @Q6ZUU2x1438x22x0 m61y9t 0:0:0:0:0:0:0:1 "POST /rest/api/latest/projects/ABC/repos/foo-repo/pull-requests/7/merge HTTP/1.1" c.a.bitbucket.scm.BaseCommand Executed /usr/bin/git rev-list --format=%H%x02%P%x02%aN%x02%aE%x02%at%n%B%n%x03END%x04 --no-walk=unsorted 2539c068ff9c7ba6ada260912e59aeef60cda8b8 2539c068ff9c7ba6ada260912e59aeef60cda8b8^@ --
2017-02-11 23:58:08,528 DEBUG [AtlassianEvent::thread-5] foo-user @Q6ZUU2x1438x22x0 m61y9t 0:0:0:0:0:0:0:1 "POST /rest/api/latest/projects/ABC/repos/foo-repo/pull-requests/7/merge HTTP/1.1" c.a.s.i.c.g.c.CachingCommitGraphSource [ABC/foo-repo[1]]: Checking RepositoryRefsChangedEvent; has adds/updates: true, is fork: false
2017-02-11 23:58:08,532 DEBUG [AtlassianEvent::thread-1] foo-user @Q6ZUU2x1438x22x0 m61y9t 0:0:0:0:0:0:0:1 "POST /rest/api/latest/projects/ABC/repos/foo-repo/pull-requests/7/merge HTTP/1.1" c.a.e.r.impl.RemoteEventListener Queueing RemoteEvent com.atlassian.stash.plugin.remote.event.StashBranchChangedRemoteEvent@4c65810f[branchName=master,repositoryProject=ABC,repositorySlug=foo-repo,committer=foo-user,sourceId=<null>,sourceUrl=<null>]
2017-02-11 23:58:08,545 DEBUG [AtlassianEvent::thread-5] foo-user @Q6ZUU2x1438x22x0 m61y9t 0:0:0:0:0:0:0:1 "POST /rest/api/latest/projects/ABC/repos/foo-repo/pull-requests/7/merge HTTP/1.1" c.a.e.r.impl.RemoteEventListener Queueing RemoteEvent com.atlassian.stash.plugin.remote.event.StashRepositoryRefsChangedEvent@5fea366[cloneLinks=[CloneLink{name=http, href=http://foo-user@localhost:7990/scm/ABC/foo-repo.git}, CloneLink{name=ssh, href=ssh://git@localhost:7999/ABC/foo-repo.git}],repositoryProject=ABC,repositorySlug=foo-repo,committer=foo-user,sourceId=<null>,sourceUrl=<null>]
2017-02-11 23:58:08,565 DEBUG [AtlassianEvent::thread-1] foo-user @Q6ZUU2x1438x22x0 m61y9t 0:0:0:0:0:0:0:1 "POST /rest/api/latest/projects/ABC/repos/foo-repo/pull-requests/7/merge HTTP/1.1" c.a.e.r.impl.RemoteEventDispatcher RemoteEvent com.atlassian.stash.plugin.remote.event.StashBranchChangedRemoteEvent@4c65810f[branchName=master,repositoryProject=ABC,repositorySlug=foo-repo,committer=foo-user,sourceId=<null>,sourceUrl=<null>] consumable by ArrayBuffer()
2017-02-11 23:58:08,565 DEBUG [AtlassianEvent::thread-5] foo-user @Q6ZUU2x1438x22x0 m61y9t 0:0:0:0:0:0:0:1 "POST /rest/api/latest/projects/ABC/repos/foo-repo/pull-requests/7/merge HTTP/1.1" c.a.e.r.impl.RemoteEventDispatcher RemoteEvent com.atlassian.stash.plugin.remote.event.StashRepositoryRefsChangedEvent@5fea366[cloneLinks=[CloneLink{name=http, href=http://foo-user@localhost:7990/scm/ABC/foo-repo.git}, CloneLink{name=ssh, href=ssh://git@localhost:7999/ABC/foo-repo.git}],repositoryProject=ABC,repositorySlug=foo-repo,committer=foo-user,sourceId=<null>,sourceUrl=<null>] consumable by ArrayBuffer()
2017-02-11 23:58:08,569 DEBUG [AtlassianEvent::thread-1] foo-user @Q6ZUU2x1438x22x0 m61y9t 0:0:0:0:0:0:0:1 "POST /rest/api/latest/projects/ABC/repos/foo-repo/pull-requests/7/merge HTTP/1.1" c.a.e.r.impl.RemoteEventListener Batch size not exceeded for com.atlassian.stash.plugin.remote.event.StashBranchChangedRemoteEvent@4c65810f[branchName=master,repositoryProject=ABC,repositorySlug=foo-repo,committer=foo-user,sourceId=<null>,sourceUrl=<null>]
2017-02-11 23:58:08,569 DEBUG [AtlassianEvent::thread-5] foo-user @Q6ZUU2x1438x22x0 m61y9t 0:0:0:0:0:0:0:1 "POST /rest/api/latest/projects/ABC/repos/foo-repo/pull-requests/7/merge HTTP/1.1" c.a.e.r.impl.RemoteEventListener Batch size not exceeded for com.atlassian.stash.plugin.remote.event.StashRepositoryRefsChangedEvent@5fea366[cloneLinks=[CloneLink{name=http, href=http://foo-user@localhost:7990/scm/ABC/foo-repo.git}, CloneLink{name=ssh, href=ssh://git@localhost:7999/ABC/foo-repo.git}],repositoryProject=ABC,repositorySlug=foo-repo,committer=foo-user,sourceId=<null>,sourceUrl=<null>]
2017-02-11 23:58:08,583 DEBUG [search-indexing:thread-1]  c.a.bitbucket.scm.BaseCommand Executed /usr/bin/git diff-tree --no-renames -r --format=%H cc2c00f9b641887ca101e1d5436f8c8ad5a8e9ad 2539c068ff9c7ba6ada260912e59aeef60cda8b8 --
2017-02-11 23:58:08,639 DEBUG [AtlassianEvent::thread-5] foo-user @Q6ZUU2x1438x22x0 m61y9t 0:0:0:0:0:0:0:1 "POST /rest/api/latest/projects/ABC/repos/foo-repo/pull-requests/7/merge HTTP/1.1" c.a.s.i.p.r.PullRequestRescopeBucketProcessor ABC/foo-repo[1]: Scheduling rescope processing
2017-02-11 23:58:08,650 DEBUG [AtlassianEvent::thread-4] foo-user @Q6ZUU2x1438x22x0 m61y9t 0:0:0:0:0:0:0:1 "POST /rest/api/latest/projects/ABC/repos/foo-repo/pull-requests/7/merge HTTP/1.1" c.a.s.i.n.h.NotificationMailer No mail configuration: discarding PullRequestMergedNotification details: PullRequestMergedNotification{user=InternalNormalUser{id=1, username=foo-user}, timestamp=Sat Feb 11 23:58:08 EST 2017, pullRequest=foo-repo:7}
2017-02-11 23:58:08,666 DEBUG [commit-indexing:thread-1]  c.a.s.i.i.DefaultCommitIndexingService [ABC/foo-repo[1]] Request for indexing received
2017-02-11 23:58:08,685 DEBUG [search-indexing:thread-1]  c.a.bitbucket.scm.BaseCommand Executed /usr/bin/git cat-file --batch-check
2017-02-11 23:58:08,686 DEBUG [commit-indexing:thread-1]  c.a.s.i.i.DefaultCommitIndexingService [ABC/foo-repo[1]] Starting indexing with indexers [com.atlassian.bitbucket.JiraKeyIndexer, com.atlassian.bitbucket.JiraKeyIndexer]
2017-02-11 23:58:08,694 DEBUG [commit-graph:thread-1] foo-user @Q6ZUU2x1438x22x0 m61y9t 0:0:0:0:0:0:0:1 "POST /rest/api/latest/projects/ABC/repos/foo-repo/pull-requests/7/merge HTTP/1.1" c.a.bitbucket.scm.BaseCommand Executed /usr/bin/git rev-list --branches --ignore-missing --topo-order --parents --tags --
2017-02-11 23:58:08,714 DEBUG [search-indexing:thread-1]  c.a.bitbucket.scm.BaseCommand Executed /usr/bin/git cat-file --batch
2017-02-11 23:58:08,717 DEBUG [search-indexing:thread-1]  c.a.b.s.i.i.BufferOnSizeAndCount Flushing remaining 1 items with file content size 6 B
2017-02-11 23:58:08,743 DEBUG [search-indexing:thread-1]  c.a.b.s.i.i.e.DefaultIndexEventWorker Indexing - Results for repository ABC/foo-repo (id: 1) is: IndexFilesResult{moreIndexingRequired=false, requestStatistics=RequestStatistics{failedDocumentIds=[], numRequests=1, numDocumentActions=1, numFailedRequests=0, numPartiallyFailedRequests=0, time=15 ms}}
2017-02-11 23:58:08,743 DEBUG [search-indexing:thread-1]  c.a.b.s.i.i.e.DefaultIndexEventWorker Indexing - Completed indexing for repository ABC/foo-repo (id: 1)
2017-02-11 23:58:08,756 DEBUG [AtlassianEvent::thread-6] foo-user @Q6ZUU2x1438x22x0 m61y9t 0:0:0:0:0:0:0:1 "POST /rest/api/latest/projects/ABC/repos/foo-repo/pull-requests/7/merge HTTP/1.1" c.a.b.i.b.a.AutoMergePullRequestListener No auto merge for ref ABC/foo-repo:refs/heads/master, reason: AUTO_MERGE_DISABLED
2017-02-11 23:58:08,757 DEBUG [AtlassianEvent::thread-5] foo-user @Q6ZUU2x1438x22x0 m61y9t 0:0:0:0:0:0:0:1 "POST /rest/api/latest/projects/ABC/repos/foo-repo/pull-requests/7/merge HTTP/1.1" c.a.e.r.impl.RemoteEventListener Queueing RemoteEvent com.atlassian.stash.plugin.remote.event.StashPullRequestMergedRemoteEvent@1682d80e[id=7,fromProjectKey=ABC,fromRepositorySlug=foo-repo,fromBranch=refs/heads/jr,toProjectKey=ABC,toRepositorySlug=foo-repo,toBranch=refs/heads/master,sourceId=<null>,sourceUrl=<null>]
2017-02-11 23:58:08,757 DEBUG [AtlassianEvent::thread-5] foo-user @Q6ZUU2x1438x22x0 m61y9t 0:0:0:0:0:0:0:1 "POST /rest/api/latest/projects/ABC/repos/foo-repo/pull-requests/7/merge HTTP/1.1" c.a.e.r.impl.RemoteEventDispatcher RemoteEvent com.atlassian.stash.plugin.remote.event.StashPullRequestMergedRemoteEvent@1682d80e[id=7,fromProjectKey=ABC,fromRepositorySlug=foo-repo,fromBranch=refs/heads/jr,toProjectKey=ABC,toRepositorySlug=foo-repo,toBranch=refs/heads/master,sourceId=<null>,sourceUrl=<null>] consumable by ArrayBuffer()
2017-02-11 23:58:08,758 DEBUG [AtlassianEvent::thread-5] foo-user @Q6ZUU2x1438x22x0 m61y9t 0:0:0:0:0:0:0:1 "POST /rest/api/latest/projects/ABC/repos/foo-repo/pull-requests/7/merge HTTP/1.1" c.a.e.r.impl.RemoteEventListener Batch size not exceeded for com.atlassian.stash.plugin.remote.event.StashPullRequestMergedRemoteEvent@1682d80e[id=7,fromProjectKey=ABC,fromRepositorySlug=foo-repo,fromBranch=refs/heads/jr,toProjectKey=ABC,toRepositorySlug=foo-repo,toBranch=refs/heads/master,sourceId=<null>,sourceUrl=<null>]
2017-02-11 23:58:08,762 DEBUG [commit-indexing:thread-1]  c.a.b.i.i.jira.idx.JiraKeyIndexer Indexed 2539c068ff9c7ba6ada260912e59aeef60cda8b8 (Matching JIRA keys: [])
2017-02-11 23:58:08,858 DEBUG [commit-indexing:thread-1]  c.a.bitbucket.scm.BaseCommand Executed /usr/bin/git rev-list --format=%H%x02%P%x02%aN%x02%aE%x02%at%n%B%n%x03END%x04 --ignore-missing --no-min-parents --stdin --
2017-02-11 23:58:08,858 DEBUG [commit-indexing:thread-1]  c.a.s.i.i.DefaultCommitIndexingService [ABC/foo-repo[1]] Scanning for deleted commits...
2017-02-11 23:58:08,872 DEBUG [commit-indexing:thread-1]  c.a.bitbucket.scm.BaseCommand Executed /usr/bin/git rev-list --format=%H%x02%P%x02%aN%x02%aE%x02%at%n%B%n%x03END%x04 --ignore-missing --no-min-parents --stdin --
2017-02-11 23:58:08,898 DEBUG [commit-indexing:thread-1]  c.a.s.i.i.DefaultCommitIndexingService [ABC/foo-repo[1]] Indexed 1 commits
```

 * Plugin version: **2.53**
 * Bitbucket Server version: **v4.13.0**
 * Screenshot of plugin configuration in your issue.

![2017-02-12_00-12-16](https://cloud.githubusercontent.com/assets/12397873/22859874/ba2f70f4-f0b9-11e6-9ae2-75e5c07ac525.png)
![2017-02-12_00-10-00](https://cloud.githubusercontent.com/assets/12397873/22859875/ba30ed9e-f0b9-11e6-8133-967d9c417d0a.png)

**Comment by tomasbjerre** [comment]
"State" is a terrible choice of label for those fields. What it does is to ignore events for pull requests if their state is **declined** or **merged**.

Perhaps "Ignore events if" is a better label?

**Comment by tomasbjerre** [comment]
I changed the label now released in 2.54
